scripts/tuple_tree_generator/tuple_tree_generator/templates/tuple_tree_gen.py.tpl

```diff
@@ -65,9 +65,47 @@ class 'enum.name'(EnumBase):
 't' = str
 ## endfor ##
 
+
+class AllMixin:
+    project = None
+
+    def set_project(self, project):
+        self.project = project
+
+    def fetch(self, artofact_name: str):
+        return self.project.fetch(artofact_name)
+
+
+class FunctionMixin:
```

Comment:
Make this mixin and the `TypeMixin` inherit from `AllMixin` to avoid duplicating code